cmd/run.go

```diff
@@ -0,0 +1,88 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"sync"
+
+	"github.com/NethermindEth/eigenda-blob-scrapper/internal/avs/eigenda"
+	"github.com/NethermindEth/eigenda-blob-scrapper/internal/config"
+	"github.com/NethermindEth/eigenda-blob-scrapper/internal/prometheus"
+	"github.com/spf13/cobra"
+)
+
+func runCommand() *cobra.Command {
+	var c *config.Config
+	return &cobra.Command{
+		Use:   "run",
+		Short: "Run the application",
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			configPath, err := cmd.Flags().GetString("config")
+			if err != nil {
+				return err
+			}
+			c, err = config.GetConfig(configPath)
+			if err != nil {
+				return err
+			}
```

Comment:
Please update the project to target Env variables first and use an optional config file. [Strive to build applications designed mainly to be deployed using environment variables](https://12factor.net/config), especially if designed to run in a container.

I understand the need to encapsulate settings as objects (`operators`). Such settings are challenging to capture in environment variables. We have a few options:

1. Have the option to pass the `config.yml` encoded as an env variable:
```
export EOE_CONFIG=$(base64 -w 0 config.yaml)
```
and then decode it and parse it in the application code.

2. Switch to json, which is easier to pass it as an env variable:
```
export EOE_CONFIG='{"operators":[{"name":"Operator1",...}],"rpcs":{...},"logLevel":"INFO"}'
```

3. Separate the settings and put the `operators` under a JSON:
```
export OPERATORS_JSON='[
  {
    "name": "Operator1",
    "address": "http://operator1.example.com",
    "blsPublicKey": ["key_part1", "key_part2"],
    "avsEnvs": ["env1", "env2"],
    "eigenDAConfig": {
      "quorums": {"1": true, "2": false}
    }
  },
  {
    "name": "Operator2",
    "address": "http://operator2.example.com",
    "blsPublicKey": ["key_part1", "key_part2"],
    "avsEnvs": ["env3", "env4"],
    "eigenDAConfig": {
      "quorums": {"1": true, "2": true}
    }
  }
]'

export EOE_RPCS_JSON='{"mainnet": "http://rpc-mainnet.example.com", "holesky": "http://rpc-holesky.example.com"}'

export EOE_LOGLEVEL="INFO"
```
4. Use indexed env variables:
```
export EOE_OPERATORS_0_NAME="Operator1"
export EOE_OPERATORS_0_ADDRESS="http://operator1.example.com"
export EOE_OPERATORS_0_BLSPUBLICKEY_0="key_part1"
export EOE_OPERATORS_0_BLSPUBLICKEY_1="key_part2"
export EOE_OPERATORS_0_AVSENVS="env1,env2"
export EOE_OPERATORS_0_EIGENDACONFIG_QUORUMS="1:true,2:false"

export EOE_OPERATORS_1_NAME="Operator2"
export EOE_OPERATORS_1_ADDRESS="http://operator2.example.com"
export EOE_OPERATORS_1_BLSPUBLICKEY_0="key_part1"
export EOE_OPERATORS_1_BLSPUBLICKEY_1="key_part2"
export EOE_OPERATORS_1_AVSENVS="env3,env4"
export EOE_OPERATORS_1_EIGENDACONFIG_QUORUMS="1:true,2:true"

export EOE_RPCS_HOLESKY="http://rpc1.example.com"
export EOE_RPCS_MAINNET="http://rpc2.example.com"

export EOE_LOGLEVEL="INFO"
```

I personally see options 2 and 3 as the most powerful ones. With 3 being the best due to the flexibility and making the codebase extendible to future settings than can be easily set as env variables. However, 2 requires minimal changes and you can still focus the configuration on a file but with the ability to pass it as an env variable (and removing the need for volumes or complex `ConfigMaps`/`Secret` configuration for K8s(

Reply:
I'm using Viper to parse the configuration. Viper supports overwriting the configuration using environment variables aligned with https://12factor.net/config. The issue is that the current configuration format is not the best for it. I propose using the config file and updating the code to support the overwrite of env variables for the next dev cycle because we need to run this service ASAP.